### ISSUE TITLE
Problem with mask for multiple objects

### ISSUE THREAD

**Issue by Nazarko12345** [learnml-today/object-detection-with-pytorch#4]
Thanks for answering.I have trained for single object detection with few classes for 15 iteration i get pretty high accuracy(0.9) and IoU for boxes (0.6).But with masks i have problem.I have 200 classes so i get output 201 (201x50x50) (+1 for background 50x50 mask size).So questions next:

How to find IoU for masks?(We have 201 output and dont know for what class to find? maybe try for first channel (background)?)
After training for 15 epochs i have a great background detection but for classes i get very bad result should i go on training?

**Comment by creotiv** [comment]
**How to find IoU for masks? (We have 201 output and don't know for what class to find? maybe try for the first channel (background)?)**
Just use MSE or MAE. But remember that edge is a more important center of the mask, so you can calculate edge accuracy and global.

**After training for 15 epochs I have a great background detection but for classes, i get very bad result should I go on training?**
That's mean that you have background class imbalance and you should minimize detection of it or loss from the detection of them.

Also, 200 classes can be too much for resnet18

**Comment by creotiv** [comment]
One more thing about metrics and losses. You should always understand what are you measuring, why and how. In such a case you can much better control training.
For example, MSE takes more care of the big differences, so it gives a much bigger gradient on start and smaller on the end. Also, it adds an additional level of nonlinearity as using a square function. MAE is uniformly distributed, so it always the same and linear. Also, you can take some instead of mean, but in that case, it will correlate with the amount of data. CrossEntropy takes more care of the small values, that's why it used for segmentation and probabilities... Etc.

So basically loss is not good or bad it task-dependent and understanding that giving more abilities.
